### PR TITLE
Add basic docs for multiprocess notifications

### PR DESCRIPTION
**Pull request by tgoyne**
A first stab at documentation for https://github.com/realm/realm-js-private/pull/391

### CODE REVIEW

**Comment by kneth** [comment]
I think we should make it clear which cases (operating systems, platforms, licenses) multiprocess notifications are supported.

**Comment by kneth** [comment]
@tgoyne I have merged #1602 which should avoid the CI failures in this PR.

Is the PR ready to be reviewed and merged?

File: docs/sync.js [hunk 1]
```diff
@@ -81,41 +116,60 @@ class Sync {
...
     get oldRealm() {}
 
     /**
-     * The change indexes for all added, removed, and modified objects in the changed Realm.
-     * This object is a hashmap of object types to arrays of indexes for all changed objects:
+     * The change indexes for all added, removed, and modified objects in the
+     * changed Realm. This object is a hashmap of object types to arrays of
+     * indexes for all changed objects:
+     *
+     * Note that deleting an object in Realm can cause other objects in the
```
Comment (kneth) [inline docs/sync.js]: Consider moving the notes to come after the example.

Comment (tgoyne) [inline docs/sync.js]: That doesn't appear to be possible with jsdoc. There's no `@endexample` or equivalent.

Comment (kneth) [reply docs/sync.js]: Sadly true, http://usejsdoc.org/tags-example.html

**Pull request merged**